stout/grpc/server.h

```diff
@@ -352,19 +386,38 @@ class Endpoint : public Synchronizable {
   }
 
   auto Dequeue() {
-    return Synchronized(
-        Wait([this](auto notify) {
-          notify_ = std::move(notify);
-          return [this]() {
-            return contexts_.empty();
-          };
-        })
-        | Lambda([this]() {
-            CHECK(!contexts_.empty());
-            auto context = std::move(contexts_.front());
-            contexts_.pop_front();
-            return context;
-          }));
+    return Repeat(
```

Comment:
It is not clear to me why the `Repeat` is moved inside the loop. I am sure you had reasons. Could I ask you to add a comment to the code?